index.js

```diff
@@ -1,39 +1,224 @@
-function CoreStore (storage) {
-  if (!(this instanceof CoreStore)) return new CoreStore(storage)
+var hypercore = require('hypercore')
+var hcrypto = require('hypercore-crypto')
+var deferred = require('deferred-random-access')
+var TinyBox = require('tinybox')
+var path = require('path')
+var LRU = require('lru')
+var { nextTick } = process
+var DKEY = 'd!' // hex discovery key  to key
+var LKEY = 'l!' // hex key to local name
+var INV_LKEY = 'L!' // local name to key
+var FEED = 'f_'
+
+function Storage (storage, opts) {
+  if (!(this instanceof Storage)) return new Storage(storage, opts)
+  if (!opts) opts = {}
+  this._db = new TinyBox(storage('db'))
+  this._storage = storage
+  this._feeds = {} // map hypercore keys to feeds
+  this._dkeys = {} // map discovery key to key for loaded feeds
+  this._lnames = {} // map local names to key for loaded feeds
+  this._delete = opts.delete
```

Comment:
```suggestion
  this._delete = opts.delete || noop
```

Reply:
Assuming it's an optional param. If it's non-optional, I think it'd be better as an explicit param.

Reply:
or throw if it's not provided